File: GuideObjC/GuideObjC/ViewControllers/PostController/NextCommentsCell.m
//
//  NextCommentsCell.m
//  GuideObjC
//
//  Created by LuzanovRoman on 19.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

#import "NextCommentsCell.h"

@interface NextCommentsCell ()
@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activityIndicatorView;
@end

@implementation NextCommentsCell

- (void)setIndicatorIsAnimated:(BOOL)indicatorIsAnimated {
    _indicatorIsAnimated = indicatorIsAnimated;
    
    if (indicatorIsAnimated) {
        [_activityIndicatorView startAnimating];
    }
    else {
        [_activityIndicatorView stopAnimating];
    }
}

- (IBAction)showNext:(id)sender {
    [self.delegate showNext];
    self.indicatorIsAnimated = true;
}
@end

File: GuideSwift/GuideSwift/ViewControllers/PostController/NextCommentsCell.swift
//
//  NextCommentsCell.swift
//  GuideSwift
//
//  Created by LuzanovRoman on 19.12.16.
//  Copyright © 2016 eigengraph. All rights reserved.
//

import UIKit

protocol NextCommentsCellDelegate: NSObjectProtocol {
    func showNext()
}

class NextCommentsCell: UITableViewCell {
    @IBOutlet weak var activityIndicatorView: UIActivityIndicatorView!
    weak var delegate: NextCommentsCellDelegate?

    var indicatorIsAnimated: Bool = false {
        didSet {
            if indicatorIsAnimated {
                activityIndicatorView.startAnimating()
            } else {
                activityIndicatorView.stopAnimating()
            }
        }
    }

    @IBAction func showNext(_ sender: AnyObject) {
        delegate?.showNext()
        indicatorIsAnimated = true
    }
}
